aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1720,21 +1732,37 @@ static Tensor& _linalg_norm_vector_out(Tensor& result, const Tensor& self, optio
     TORCH_CHECK(unique_dims, "Expected dims to be different, got this instead: (", dim, ")");
   }
   if (!case_was_overridden) {
-    if (opt_dtype.has_value()) {
-      result_ = at::norm(self.to(opt_dtype.value()), opt_ord, dim, keepdim);
+    if (flatten) {
+      if (opt_dtype.has_value()) {
+        result_ = at::norm(self.to(opt_dtype.value()), opt_ord.value_or(2));
```

Comment:
Why is it `opt_ord.value_or(2)` here but not in the branch where `flatten` is false?

Reply:
Does torch.norm handle cast to dtype implicitly, at least for cuda fp16->fp32? Yes, it does, so that's another perf penalty here.